pydantic/_internal/_fields.py

```diff
@@ -259,7 +260,13 @@ def collect_model_fields(  # noqa: C901
 
     # https://docs.python.org/3/howto/annotations.html#accessing-the-annotations-dict-of-an-object-in-python-3-9-and-older
     # annotations is only used for finding fields in parent classes
-    annotations = cls.__dict__.get('__annotations__', {})
+    if sys.version_info >= (3, 14):
+        from annotationlib import Format, get_annotations
+
+        annotations = get_annotations(cls, format=Format.FORWARDREF)
+    else:
+        annotations = cls.__dict__.get('__annotations__', {})
```

Comment:
Think there's a call to `_typing_extra.safe_get_annotations(cls)` still to be added here?

```suggestion
    annotations = _typing_extra.safe_get_annotations(cls)
```